src/platform/sys.c

```diff
@@ -11,3 +11,21 @@ MVMuint32 MVM_platform_cpu_count(void) {
 
     return count;
 }
+
+void MVM_platform_uname(MVMThreadContext *tc, MVMObject *result) {
+    uv_utsname_t uname;
+    int r = uv_os_uname(&uname);
+    if (r != 0)
+        MVM_exception_throw_adhoc(tc, "Unable to uname: %s", uv_strerror(r));
+    if (REPR(result)->ID != MVM_REPR_ID_VMArray || !IS_CONCRETE(result) ||
+            ((MVMArrayREPRData *)STABLE(result)->REPR_data)->slot_type != MVM_ARRAY_STR) {
+        MVM_exception_throw_adhoc(tc, "uname needs a concrete string array.");
+    }
+
+    MVMROOT(tc, result, {
+        MVM_repr_bind_pos_s(tc, result, 0, MVM_string_utf8_decode(tc, tc->instance->VMString, uname.sysname, strlen((char *)uname.sysname)));
```

Comment:
This has a subtle problem: we root the local `result`, but if we're passing it as an argument then it will be potentially read and placed on the stack or into a register. Then the `decode` can allocate, and leave the register/stack location that `result` was earlier read into outdated, leading to problems. So the two calls need to be split over two lines, with a temporary.